tsconfig.build.json

```diff
@@ -0,0 +1,9 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "noEmit": false,
+    "outDir": "dist",
+    "emitDeclarationOnly": true
+  },
+  "include": ["lib"]
```

Comment:
When we're building we really only want to include `lib`... but when we're doing general type checking we'd like to include tests too. You can't really do includes/excludes from the CLI while also providing a config so it becomes necessary to have two configs. So, yeah. That's why this is here. It also ensures if you ever run `yarn tsc` it won't accidentally spit out files.